Confirm MacOS support

Since, the build works well on Linux doesn't mean it would work fine on MacOS too. It would be better if someone confirms on this.

```
Ayush-MacBook-Pro:advCalc-master Ayush$ make
c++ -c -std=c++14  main.cpp
In file included from main.cpp:7:
In file included from ./calcParser.hpp:7:
./str.hpp:15:17: error: functions that differ only in their return type cannot be overloaded
extern uint64_t strlen(constStr s);
       ~~~~~~~~ ^
/usr/include/string.h:82:9: note: previous declaration is here
size_t   strlen(const char *__s);
~~~~~~   ^
1 error generated.
make: *** [main.o] Error 1
```

this was the error when I gave command make..

I get this error while building this project.
```
CMake Error at src/calc_ui_qt/CMakeLists.txt:5 (find_package):
  By not providing "FindQt5.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5" with any of
  the following names:

    Qt5Config.cmake
    qt5-config.cmake

  Add the installation prefix of "Qt5" to CMAKE_PREFIX_PATH or set "Qt5_DIR"
  to a directory containing one of the above files.  If "Qt5" provides a
  separate development package or SDK, be sure it has been installed.
```

Guide me further to build this, I've already installed qt5 
```brew install qt5```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,13 @@ set(CMAKE_CXX_STANDARD 14)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/modules")
 
+if (EXISTS "${ROOT}/usr/local/opt/readline/include")
+  include_directories(${ROOT}/usr/local/opt/readline/include)
+  message(STATUS "Found readline Headers")
+else()
+  message(STATUS "Sorry")
+endif()
+
 # message(STATUS "some ${CMAKE_SOURCE_DIR}")
 # message(STATUS "some ${PROJECT_SOURCE_DIR}")
 

--- a/src/calcOptr.cpp
+++ b/src/calcOptr.cpp
@@ -74,8 +74,8 @@ void Operator::operator=(const Operator &x) {
 }
 
 static str_hash generateHashKey(constStr s, ulong start = 0, ulong end = 0) {
-  register constStr x = s = s + start;
-  register str_hash hash = 0;
+  constStr x = s = s + start;
+  str_hash hash = 0;
   if (start == end)
     while (*x)
       hash = 127 * hash + *(x++);

--- a/src/input_bindings.cpp
+++ b/src/input_bindings.cpp
@@ -1,5 +1,7 @@
-#include <malloc.h>
+#include <stdio.h>
 #include <readline/readline.h>
+#include <readline/chardefs.h>
+#include "str.hpp"
 
 static bool trim_spaces = false;
 
@@ -27,6 +29,8 @@ static int space_pressed(int x, int input_char) {
   }
   return x;
 }
+
+#if 0
 /* under dev code, edit/use with precaution */
 static int rubout_pressed(int x, int input_char) {
   if (trim_spaces == true) {
@@ -72,6 +76,8 @@ static int rubout_pressed(int x, int input_char) {
   return rl_delete(x, input_char);
 }
 /* **************************************** */
+#endif // 0
+
 static int do_nothing(int x, int input_char) { return x + input_char; }
 
 void init_readline() {
@@ -79,7 +85,9 @@ void init_readline() {
   rl_bind_key(TAB, do_nothing);
   rl_bind_key(RETURN, return_pressed);
   rl_bind_key(NEWLINE, return_pressed);
+#if 0
   /* under dev line, use/edit with precaution */
   rl_bind_key(RUBOUT, rubout_pressed);
   /* **************************************** */
+#endif //0
 }

--- a/src/str.cpp
+++ b/src/str.cpp
@@ -2,6 +2,7 @@
 #include <ctype.h>
 #include <math.h>
 
+#if !defined(_STRING_H) && !defined(STRING_H) && !defined(_STRING_H_)
 uint64_t strlen(constStr s) {
   int l = 0;
   for (; s[l]; l++)
@@ -70,6 +71,7 @@ str strncat(str s1, constStr s2, uint64_t l) {
     s1[i] = 0;
   return s1;
 }
+#endif // STRING_H
 
 bool isidentifier(constStr s) {
   while (isalpha(*s) || *s == '_')

--- a/src/str.hpp
+++ b/src/str.hpp
@@ -12,7 +12,7 @@ inline void skipSpace(constStr &s) {
   isspace(*s) ? ++s : s;
 }
 
-#if !defined(_STRING_H) && !defined(STRING_H)
+#if !defined(_STRING_H) && !defined(STRING_H) && !defined(_STRING_H_)
 extern uint64_t strlen(constStr s);
 extern int strcmp(constStr s1, constStr s2);
 extern int strncmp(constStr s1, constStr s2, uint64_t l);

--- a/src/str.hpp
+++ b/src/str.hpp
@@ -28,10 +28,6 @@ extern bool isidentifier(constStr s);
 
 extern signed char strToNum(constStr *a, double &x, datatype d);
 
-#ifdef ANS_CMD
-extern signed char separate_ans(constStr a, unsigned long &i, unsigned long &x);
-#endif
-
 extern str trimSpaces(constStr s);
 
 #endif // CALC_STR_H

--- a/src/str.hpp
+++ b/src/str.hpp
@@ -12,6 +12,7 @@ inline void skipSpace(constStr &s) {
   isspace(*s) ? ++s : s;
 }
 
+#if !defined(_STRING_H) && !defined(STRING_H)
 extern uint64_t strlen(constStr s);
 extern int strcmp(constStr s1, constStr s2);
 extern int strncmp(constStr s1, constStr s2, uint64_t l);
@@ -21,6 +22,7 @@ extern char* strcpy(str s1, constStr s2);
 extern char* strncpy(str s1, constStr s2, uint64_t l);
 extern char* strcat(str s1, constStr s2);
 extern char* strncat(str s1, constStr s2, uint64_t l);
+#endif
 
 extern bool isidentifier(constStr s);
 

